### PR TITLE
Pricing page rework: Display additional legacy product cards

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR updates the product and bundle lists to display additional legacy product card(s) if they are currently owned/active on the site (for a signed-in user)

#### Proposed Changes

* Add icons for Security and Complete bundles
* Move `getCtaLabel` to `useStoreItemInfo` hook for reusability
* Show "All bundles" section when a legacy item is owned by a user
* Fix product name i18n when name is JSX Element

#### Testing Instructions

- Create a JN site and connect Jetpack
- Purchase a legacy product for the site e.g. Security Daily using this link after replacing `:site` with the actual site URL `https://wordpress.com/checkout/:site/jetpack_security_daily`
- Do any one of these
    - Open Calypo Blue and Green in two separate tabs
        - Click on Jetpack Cloud Live link below and wait for the redirect to complete, then goto `/pricing/:site?flags=jetpack/pricing-page-rework-v1`
        - Click on Calypso Live link below and wait for the redirect to complete, then goto `/plans/:site?flags=jetpack/pricing-page-rework-v1`
    - or boot up this PR 
        - Run `git fetch && git checkout fix/jp-pricing-rework/legacy-products`
        - Run `yarn start` in one terminal window and `yarn start-jetpack-cloud-p` in another one.
        - Open this in one tab `http://calypso.localhost:3000/plans/:site?flags=jetpack/pricing-page-rework-v1`
        - Open this in another one `http://jetpack.cloud.localhost:3001/pricing/:site?flags=jetpack/pricing-page-rework-v1&view=bundles`
- Confirm these steps for both Calypso Blue and Green
    - Confirm that you see the new pricing UI
    - Confirm that you see the "Backup Daily" card in "All products" section
    - Goto Bundles tab
    - Confirm that you see "All bundles" section
    - Confirm that you see "Security Daily" card in the list
- Goto `/pricing/?flags=jetpack/pricing-page-rework-v1` for Jetpack Cloud
- Confirm that you don't see "All bundles" section

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Cpmpletes: 1202796695664022-as-1202924222500997/f

Screenshot 2022-09-06 at 3 36 01 PM
Screenshot 2022-09-06 at 6 29 06 PM
Screenshot 2022-09-06 at 6 43 07 PM
Screenshot 2022-09-06 at 6 43 22 PM
Screenshot 2022-09-06 at 6 43 45 PM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202924222500997

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/product-store/featured-item-card/index.tsx [hunk 1]
```diff
@@ -49,9 +49,9 @@ export const FeaturedItemCard: React.FC< FeaturedItemCardProps > = ( {
 									href="#"
 									plain
 								>
-									{ translate( 'More about %(product)s', {
-										args: {
-											product: title,
+									{ translate( 'More about {{productName/}}', {
+										components: {
+											productName: <>{ title }</>,
```
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-store/featured-item-card/index.tsx:54]: It's possible that `title` can be React element. In that case the link text does not show up correctly.

File: client/my-sites/plans/jetpack-plans/product-store/bundle-list/index.tsx [hunk 2]
```diff
@@ -93,10 +71,58 @@ export const BundlesList: React.FC< BundlesListProps > = ( {
...
+		[ otherItems, popularItems ]
+	);
+
 	return (
 		<div className="jetpack-product-store__bundles-list">
 			<MostPopular heading={ translate( 'Most popular bundles' ) } items={ mostPopularItems } />
 			<SeeAllFeatures />
+
+			{ /* Show All items only if there is something in otherItems */ }
+			{ otherItems.length ? (
```
Comment (jkguidaven) [inline client/my-sites/plans/jetpack-plans/product-store/bundle-list/index.tsx:85]: Is it possible to make this chunk of codes reusable component so we can also use this in the product list?

Comment (manzoorwanijk) [reply client/my-sites/plans/jetpack-plans/product-store/bundle-list/index.tsx:85]: Yes, that's a good observation. I even started to do that in this PR but then decided to do it in a separate PR.

**Review by jkguidaven — APPROVED** [review]
LGTM.. tested and it is working as expected. 👍 good job as always.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7514265

Thank you @manzoorwanijk for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7514265).

**Pull request merged**